babel-watch.js

```diff
@@ -222,43 +222,42 @@ function generateTempFilename() {
   ].join(''));
 }
 
-function handleFileLoad(filename, callback) {
+async function handleFileLoad(filename) {
   const cached = cache[filename];
   if (cached) {
     const stats = fs.statSync(filename);
     if (stats.mtime.getTime() === cached.mtime) {
-      callback(cache[filename].code, cache[filename].map);
-      return;
+      return [cache[filename].code, cache[filename].map];
     }
   }
   if (!shouldIgnore(filename)) {
-    compile(filename, (err, result) => {
-      debugCompile('Compiled file: %s. Success? %s', filename, !err);
-
-      if (!result && !err) err = new Error('No Result from Babel for file: ' + filename);
-      if (err || !result) {
-        // Intentional ignore
-        if (err instanceof IgnoredFileError) {
-          ignored[filename] = true;
-          debugCompile('File %s ignored due to extension or intentional ignore rule.', filename);
-          return callback();
-        }
-        console.error('Babel compilation error', err.stack);
-        errors[filename] = true;
-        return;
+    try {
+      const result = await compile(filename)
+      if (!result) {
+        throw new Error('No Result from Babel for file: ' + filename);
       }
+      debugCompile('Compiled file: %s. Success? true', filename);
       const stats = fs.statSync(filename);
       cache[filename] = {
         code: result.code,
         map: result.map,
         mtime: stats.mtime.getTime(),
       };
       delete errors[filename];
-      callback(result.code, result.map);
-    });
-  } else {
-    callback();
+      return [result.code, result.map];
+    } catch (err) {
+      debugCompile('Compiled file: %s. Success? false', filename);
+      // Intentional ignore
+      if (err instanceof IgnoredFileError) {
+        ignored[filename] = true;
+        debugCompile('File %s ignored due to extension or intentional ignore rule.', filename);
+        return [];
+      }
+      console.error('Babel compilation error', err.stack);
+      errors[filename] = true;
```

Comment:
This branch didn't fire a callback, now it does resolve. Could be for the best (or not)

Reply:
Yeah it appears this would lead to a printed error but nothing else. Now it will lead to a full process exit. I'm not sure this is desired.